pkg/apis/gcp/types_backupbucket.go

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package gcp
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupBucketConfig represents the configuration for a backup bucket.
+type BackupBucketConfig struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Immutability defines the immutability config for the backup bucket.
+	Immutability ImmutableConfig `json:"immutability"`
+}
+
+// ImmutableConfig represents the immutability configuration for a backup bucket.
+type ImmutableConfig struct {
```

Comment:
Please adapt this [file](https://github.com/gardener/gardener-extension-provider-gcp/blob/master/example/30-backupbucket.yaml) for this change.